tests/supply_chain/test_action_reset.py

```diff
@@ -0,0 +1,790 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import random
+import unittest
+from collections import defaultdict
+from typing import Dict, List
+
+import numpy as np
+
+from maro.simulator.scenarios.supply_chain import (
+    ConsumerAction,
+    ConsumerUnit,
+    FacilityBase,
+    ManufactureAction,
+    ManufactureUnit,
+    RetailerFacility,
+    StorageUnit,
+    WarehouseFacility,
+)
+from maro.simulator.scenarios.supply_chain.business_engine import SupplyChainBusinessEngine
+
+from tests.supply_chain.common import SKU1_ID, SKU3_ID, build_env, get_product_dict_from_storage
+
+
+class MyTestCase(unittest.TestCase):
+    """
+    . consumer unit test
+    . distribution unit test
+    . manufacture unit test
+    . seller unit test
+    . storage unit test
+    """
+
+    def test_env_reset_with_none_action(self) -> None:
+        """test_env_reset_with_none_action"""
+        env = build_env("case_05", 500)
+        be = env.business_engine
+        assert isinstance(be, SupplyChainBusinessEngine)
+
+        env.step(None)
+
+        supplier_1: FacilityBase = be.world._get_facility_by_name("Supplier_SKU1")
+        warehouse_1 = be.world._get_facility_by_name("Warehouse_001")
+        Store_001: FacilityBase = be.world._get_facility_by_name("Store_001")
+
+        consumer_unit: ConsumerUnit = warehouse_1.products[SKU3_ID].consumer
+        storage_unit: StorageUnit = supplier_1.storage
+        Store_001.products[SKU3_ID].seller
+        supplier_1.products[SKU1_ID].manufacture
+        supplier_1.distribution
+
+        consumer_nodes = env.snapshot_list["consumer"]
+        storage_nodes = env.snapshot_list["storage"]
+        seller_nodes = env.snapshot_list["seller"]
+        manufacture_nodes = env.snapshot_list["manufacture"]
+        distribution_nodes = env.snapshot_list["distribution"]
+
+        consumer_features = (
```

Comment:
you can put these features in common, no need to re-claim them in each test function